pkgs/development/python-modules/mocket/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib, buildPythonPackage, fetchPypi, pythonOlder, isPy27
+, decorator
+, importlib-metadata
+, python
+, python_magic
+, six
+, urllib3 }:
+
+buildPythonPackage rec {
+  pname = "mocket";
+  version = "3.8.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1zj2p20blf4p071kcs86h2c9sc964x6sahff60gg7y4d2c0iq3rs";
+  };
+
+  patchPhase = ''
+    substituteInPlace requirements.txt --replace "python-magic==0.4.18" "python-magic"
+    substituteInPlace requirements.txt --replace "urllib3==1.25.10" "urllib3"
+    substituteInPlace setup.py --replace "setup_requires=[\"pipenv\"]" "setup_requires=[]"
+  '';
+
+  propagatedBuildInputs = [
+    decorator
+    python_magic
+    urllib3
+    six
+  ] ++ lib.optionals (isPy27) [ six ];
+
+  # Pypi has no runtests.py, github has no requirements.txt. No way to test, no way to install.
+  doCheck = false;
```

Comment:
Maybe file an upstream issue?

Also, it looks like the ``runtest.py`` basically just runs pytest, and the tests are included in the PyPi sdist. So maybe just try running ``pytestCheckHook``?

Reply:
generally a `tox.ini` or the `test_requires` section of setup.py will mention what is needed.

However, this does a non-standard construction. Needs `xxhash` `aiohttp` and `async-timeout`
https://github.com/mindflayer/python-mocket/blob/fb24872ae2a530dad0e645e3e9c39dcea29e999d/runtests.py#L22

Reply:
Testing mocket requires packaging of pook, which requires mocket for testing.

Boom. Infinite loop.

I only ever wanted to fixup those packages, not maintain them :disappointed: